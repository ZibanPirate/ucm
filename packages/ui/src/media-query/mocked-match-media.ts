export const mockedMatchMedia =
  (assignCallback: (callback: (e: MediaQueryListEvent) => void) => void) => (): MediaQueryList =>
    ({
      addEventListener: (_event: string, callback: (e: MediaQueryListEvent) => void) => {
        assignCallback(callback);
      },
      removeEventListener: () => null,
      matches: true,
    } as unknown as MediaQueryList);
