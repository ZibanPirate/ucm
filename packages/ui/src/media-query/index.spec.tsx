import { render } from "@testing-library/react";

import { MediaQuery } from ".";
import { mockedMatchMedia } from "./mocked-match-media";

describe(`Testing component '${MediaQuery.name}' :`, () => {
  it(`should render with 4 div elements then hide responsively`, () => {
    let callbackMock: (e: MediaQueryListEvent) => void = () => null;
    window.matchMedia = mockedMatchMedia((callback) => (callbackMock = callback));
    const { container } = render(
      <MediaQuery query="test-query">
        {[1, 2, 3, 4].map((index) => (
          <div key={index}>{index}</div>
        ))}
      </MediaQuery>,
    );
    expect(container).toMatchSnapshot("shown");

    callbackMock({ matches: false } as unknown as MediaQueryListEvent);
    expect(container).toMatchSnapshot("hidden");
  });
});
