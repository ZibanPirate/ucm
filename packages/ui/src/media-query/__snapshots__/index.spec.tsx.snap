// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component 'MediaQuery' : should render with 4 div elements then hide responsively: hidden 1`] = `<div />`;

exports[`Testing component 'MediaQuery' : should render with 4 div elements then hide responsively: shown 1`] = `
<div>
  <div>
    1
  </div>
  <div>
    2
  </div>
  <div>
    3
  </div>
  <div>
    4
  </div>
</div>
`;
